builtins.go

```diff
@@ -853,14 +852,23 @@ func builtinUglyObjectFlatMerge(i *interpreter, trace TraceElement, x value) (va
 					bindings: simpleObj.upValues,
 				},
 			}
-			// another ugliness - we just take the locals of our last object,
-			// we assume that the locals are the same for each of merged objects
-			locals = simpleObj.locals
 		}
+		anyObj = simpleObj
+	}
+
+	var locals []objectLocal
+	var localUpValues bindingFrame
+	if len(objarr.elements) > 0 {
+		simpleObj := anyObj
```

Comment:
could delete this line and just use anyObj

Reply:
Thanks, fixed